dev-python/pydispatcher/pydispatcher-2.0.1.recipe

```diff
@@ -1,25 +1,63 @@
-DESCRIPTION="
-Multi-producer-multi-consumer signal dispatching mechanism in python.
-"
+SUMMARY="Multi-producer-multi-consumer signal dispatching mechanism in Python"
+DESCRIPTION="PyDispatcher provides the Python programmer with a \
+multiple-producer-multiple-consumer, signal-registration and routing infrastructure for \
+use in multiple contexts. The mechanism of PyDispatcher started life as a highly rated \
+recipe in the Python Cookbook. The SourceForge project aims to include various \
+enhancements to the recipe developed during use in various applications."
 HOMEPAGE="http://pydispatcher.sourceforge.net/"
-SOURCE_URI="http://downloads.sourceforge.net/project/pydispatcher/pydispatcher/2.0.1/PyDispatcher-2.0.1.tar.gz"
+COPYRIGHT="2001-2006, Patrick K. O'Brien and Contributors"
+LICENSE="BSD (3-clause)"
 REVISION="1"
-STATUS_HAIKU="stable"
-DEPEND="dev-python/setuptools >= 0.6"
-CHECKSUM_MD5="8b1695ed6f5bdbda8fbcde82b62d505c"
+SOURCE_URI="http://downloads.sourceforge.net/project/pydispatcher/PyDispatcher-$portVersion.tar.gz"
+CHECKSUM_SHA256="a126561446cd1c9fdee0d43a32cae4413831dc017a64baf2f2fecd1597203898"
+SOURCE_DIR="PyDispatcher-$portVersion"
 
-BUILD()
-{
-	cd PyDispatcher-2.0.1
-	python setup.py build
-}
+ARCHITECTURES="any"
+
+PROVIDES="
+	pydispatcher = $portVersion
+	"
+REQUIRES="
+	haiku
+	"
+
+BUILD_REQUIRES="
+	haiku_devel
+	"
+
+PYTHON_PACKAGES=(python)
+PYTHON_VERSIONS=(2.7)
+for i in "${!PYTHON_PACKAGES[@]}"; do
+pythonPackage=${PYTHON_PACKAGES[i]}
+pythonVersion=${PYTHON_VERSIONS[$i]}
+eval "PROVIDES_${pythonPackage}=\"\
+	${portName}_$pythonPackage = $portVersion\
+	\"; \
+REQUIRES_$pythonPackage=\"\
+	haiku\n\
+	cmd:python$pythonVersion\
+	\""
+BUILD_REQUIRES="$BUILD_REQUIRES
+	setuptools_$pythonPackage"
+BUILD_PREREQUIRES="$BUILD_PREREQUIRES
+	cmd:python$pythonVersion"
+done
 
 INSTALL()
 {
-	cd PyDispatcher-2.0.1
-	python setup.py install --root=${DESTDIR}
+	for i in "${!PYTHON_PACKAGES[@]}"; do
+		pythonPackage=${PYTHON_PACKAGES[i]}
+		pythonVersion=${PYTHON_VERSIONS[$i]}
 
-}
+		python=python$pythonVersion
+		installLocation=$prefix/lib/$python/vendor-packages/
+		export PYTHONPATH=$installLocation:$PYTHONPATH
+		mkdir -p $installLocation
+		rm -rf build
+		$python setup.py build install \
+			--root=/ --prefix=$prefix
 
-LICENSE="BSD (3-clause)"
-COPYRIGHT="2001-2006, Patrick K. O'Brien and Contributors"
+		packageEntries  $pythonPackage \
+			$prefix/lib/python*
+	done
+}
```

Comment:
The last line is missing a LineFeed. Could you please add one here?
No need to add en empty line. Editing the file with vim and saving it will automatically add the newline.
If you have `ed` then it's even easier:
```bash
echo wq | ed pydispatcher-2.0.1.recipe
```